File: 10-09-2026/raph.c
#include<stdio.h>
#include<math.h>
double raph(double x){
	return (x-((exp(x)-2)/(exp(x))));
}
int main(){
	double x=1;
	for (int i=0;i<20;i++){
		printf("%lf\n",x);
		x=raph(x);
	}
	printf("Newton raphson gave %lf\n",x)
}

File: 10-09-2026/raph.py
import matplotlib.pyplot as plt
import numpy as np
import subprocess
import shlex
import math
def func(x):
    return math.exp(x)-2
def raph(x):
    return  (x-(math.exp(x)/(math.exp(x)-2)))
x=1;
x1=np.linspace(-3,2,1000);
y1=(np.exp(x1)-2)
plt.plot(x1,y1)
for i in range (20):
    plt.plot(x,func(x),"o")
    x=raph(x)
plt.savefig("raph.pdf")
subprocess.run(shlex.split("termux-open raph.pdf"))
